packages/persist-web-storage/src/index.ts

```diff
@@ -77,11 +78,152 @@ const reatomPersistWebStorage = (name: string, storage: Storage) => {
+      if (memCache.has(key)) {
+        const prev = memCache.get(key)
+        ctx.schedule(() => memCache.set(key, prev!), -1)
+      }
+      memCache.delete(key)
+      ctx.schedule(() =>
+        channel.postMessage({ key, rec: null } satisfies BroadcastMessage),
+      )
+    },
+    subscribe(ctx, key, cb) {
+      const memCache = ctx.get(memCacheAtom)
+      const handler = (event: MessageEvent<BroadcastMessage>) => {
+        const { key: messageKey, rec } = event.data
+        if (messageKey === key) {
+          if (rec === null) {
+            memCache.delete(key)
+            cb()
+          } else {
+            if (rec.id !== memCache.get(key)?.id) {
+              memCache.set(key, rec)
+              cb()
+            }
+          }
+        }
+      }
+      channel.addEventListener('message', handler, false)
+      return () => channel.removeEventListener('message', handler, false)
+    },
+  })
+}
+
+const reatomPersistIndexedDb = (dbName: string, channel: BroadcastChannel) => {
+  const memCacheAtom = atom(
+    new Map<string, PersistRecord>(),
+    `withIndexedDb._memCacheAtom`,
+  )
+
+  const store = createStore(dbName, 'atoms')
+
+  return reatomPersist({
+    name: 'withIndexedDb',
+    get(ctx, key) {
+      const memCache = ctx.get(memCacheAtom)
+      return memCache.get(key) ?? null
+    },
+    set(ctx, key, rec) {
+      const memCache = ctx.get(memCacheAtom)
+      memCache.set(key, rec)
+      ctx.schedule(async () => {
+        await set(key, rec, store)
+        channel.postMessage({ key, rec } satisfies BroadcastMessage)
+      })
+    },
+    clear(ctx, key) {
+      const memCache = ctx.get(memCacheAtom)
+      memCache.delete(key)
+      ctx.schedule(async () => {
+        await del(key, store)
+        channel.postMessage({ key, rec: null } satisfies BroadcastMessage)
+      })
+    },
+    subscribe(ctx, key, cb) {
+      ctx.schedule(async (ctx) => {
+        const data = await get(key, store)
+        const memCache = ctx.get(memCacheAtom)
+        memCache.set(key, data)
+        cb()
+      })
+
+      const handler = (event: MessageEvent<BroadcastMessage>) => {
+        const { key: messageKey, rec } = event.data
+        if (messageKey === key) {
+          const memCache = ctx.get(memCacheAtom)
+          if (rec === null) {
+            memCache.delete(key)
+            cb()
+          } else {
+            if (rec.id !== memCache.get(key)?.id) {
+              memCache.set(key, rec)
+              cb()
+            }
+          }
+        }
+      }
+
+      channel.addEventListener('message', handler)
+      return () => channel.removeEventListener('message', handler)
+    },
+  })
+}
+
 export const withLocalStorage = reatomPersistWebStorage(
   'withLocalStorage',
   globalThis.localStorage,
 )
+
 export const withSessionStorage = reatomPersistWebStorage(
   'withSessionStorage',
   globalThis.sessionStorage,
 )
+
+export const withBroadcastChannel = (
```

Comment:
The API should not be as a chain of functions calls. Please, merge all needed options to one config parameter for one adapter call (`pipe(withBroadcastChannel()())` -> `pipe(withBroadcastChannel())`)